Frontend/ui-customer/src/Screens/CustomerUtils/ReportIssueChatScreen/Controller.purs

```diff
@@ -0,0 +1,475 @@
+            exit (SelectIssueOption (state { data { chatConfig { messages = messages', suggestionsList = []  }, selectedOption = Just selectedOption } }))
+    Nothing -> do
+      _ <- pure $ toast "Can't find option"
+      continue state
+
+eval (ChatViewActionController (ChatView.BackPressed)) state = do
+  _ <- pure $ hideKeyboardOnNavigation true
+  continueWithCmd state [do
+    pure $ BackPressed
+  ]
+
+eval (ChatViewActionController (ChatView.OnImageClick url)) state = do
+  continue state { data  { viewImageState { image = url, imageName = Nothing } }
+                 , props { showViewImageModel = true } }
+
+eval (ChatViewActionController (ChatView.EnableSuggestions)) state =
+  continue state { data { chatConfig { enableSuggestionClick = true } } }
+
+eval (SendMessage message toggleSubmitComp) state = do
+  let messages'       = snoc state.data.chatConfig.messages message
+  let showSubmitComp' = if toggleSubmitComp then not state.props.showSubmitComp else state.props.showSubmitComp
+  continueWithCmd state { data  { chatConfig { messages = messages' } }
+                        , props { showSubmitComp = showSubmitComp' } } [do
+    _ <- scrollToEnd (getNewIDWithTag "ChatScrollView") true
+    pure NoAction
+  ]
+
+eval (UpdateState updatedState) state = do
+  continue updatedState
+
+eval (PrimaryEditTextActionController (PrimaryEditText.TextChanged _ text)) state =
+  continue state { data { messageToBeSent = if (ST.length text >= 256) then state.data.messageToBeSent else text } }
+
+eval (ConfirmCall (PrimaryButton.OnClick)) state =
+  case find (\x -> (state.props.showCallSupportModel && x.label == "CALL_SUPPORT") ||(state.props.showCallDriverModel && x.label == "CALL_DRIVER")) state.data.options of
+    Just selectedOption -> do
+      let messages' = snoc state.data.chatConfig.messages (ChatView.makeChatComponent selectedOption.option "Customer" (getCurrentUTC "") )
+      if state.props.showCallDriverModel
+      then
+        exit $ CallDriver state { data { chatConfig { messages = messages', suggestionsList = [] }, selectedOption = Just selectedOption }, props { isPopupModelOpen = false, showCallDriverModel = false, showCallSupportModel = false } }
+      else
+        exit $ CallSupport state { data { chatConfig { messages = messages', suggestionsList = [] }, selectedOption = Just selectedOption }, props { isPopupModelOpen = false, showCallDriverModel = false, showCallSupportModel = false } }
+    Nothing -> do
+      _ <- pure $ toast "Can't find option"
+      continue state
+
+eval (CancelCall (PrimaryButton.OnClick)) state =
+  continue state { props { isPopupModelOpen = false, showCallDriverModel = false, showCallSupportModel = false } }
+
+eval InitializeCallback state = continue state { props {initalizedCallbacks = true}}
+
+eval _ state =
+  continue state
+
+data Result = Result Boolean
+
+downloadInvoicePDF :: ReportIssueChatScreenState -> Unit
+downloadInvoicePDF state = 
+  case state.data.selectedRide of
```

Comment:
use maybe function here, instead of case match. 
refer to the following link for more context : 
https://pursuit.purescript.org/packages/purescript-maybe/6.0.0/docs/Data.Maybe#v:maybe